npm_and_yarn/helpers/lib/npm7/subdependency-updater.js

```diff
@@ -2,33 +2,42 @@ const fs = require("fs");
 const path = require("path");
 const npm = require("npm7");
 const Arborist = require("@npmcli/arborist");
+const { formatErrorMessage } = require("./helpers");
 
-async function updateDependencyFile(directory, lockfileName, dependencies) {
-  const readFile = (fileName) =>
-    fs.readFileSync(path.join(directory, fileName)).toString();
-
+const install = async (directory, lockfileName, dependencies) => {
   await new Promise((resolve) => {
     npm.load(resolve);
   });
 
   const arb = new Arborist({
     ...npm.flatOptions,
     path: directory,
-    packageLockOnly: true,
+    packageLockOnly: false,
```

Comment:
This is a downgrade from what we have, results in downloading `node_modules` but seems to be the only way to get resolution errors from arborist.

The benefit of the new behaviour is that arborist only seems to bail when the updated dependencies are not resolvable and leaves existing lockfile entries as is. We'd need to rejig a bunch of test to account for this so leaving it disable for now.